src/App.jsx

```diff
@@ -27,14 +30,51 @@ function App() {
     // Total retirement savings
     const totalRetirementSavings = futureValueCurrentSavings + futureValueContributions
 
-    // 4% rule calculations
-    const annualWithdrawal = totalRetirementSavings * 0.04
+    // Calculate withdrawals based on selected strategy
+    let annualWithdrawal = 0
+    let strategyName = ''
+    let strategyDescription = ''
+
+    switch (withdrawalStrategy) {
+      case '4percent': {
+        annualWithdrawal = totalRetirementSavings * 0.04
+        strategyName = '4% Rule'
+        strategyDescription = 'The 4% rule suggests that you can safely withdraw 4% of your retirement portfolio each year without running out of money for at least 30 years. This calculation assumes your investments continue to grow during retirement.'
+        break
+      }
+      case '1overN': {
+        annualWithdrawal = totalRetirementSavings / retirementDurationNum
+        strategyName = '1/N Withdrawal'
+        strategyDescription = `The 1/N strategy divides your total retirement savings by the number of years you expect to be retired (${retirementDurationNum} years). This provides a simple way to ensure your money lasts exactly as long as planned, but doesn't account for investment growth during retirement.`
+        break
+      }
+      case 'cape': {
```

Comment:
⚖️ **Midlevel Developer**

For the calculation logic in `src/App.jsx`, I recommend extracting the withdrawal calculations into their own pure functions. This will make it easier to write unit tests for each strategy and follow the Single Responsibility Principle (SRP) from SOLID. At my last company, we always separated calculation logic from UI code. You might consider a `withdrawalStrategies.js` file exporting each strategy as a method. That way, you can add more strategies in the future and keep the codebase maintainable.